packages/amplify_flutter/lib/categories/amplify_auth_category.dart

```diff
@@ -115,27 +183,47 @@ class AuthCategory {
         : throw _pluginNotAddedException("Auth");
   }
 
+  /// Returns an [AuthUser] or plugin-specific subclass.
+  ///
+  /// Throws an [AuthException], or possibly a subclass such as [SignedOutException].
+  /// Other AuthException subclasses are possible as a result of network issues,
+  /// misconfiguration, internal service errors or other less common reasons.
   Future<AuthUser> getCurrentUser() {
     var request = AuthUserRequest();
     return plugins.length == 1
         ? plugins[0].getCurrentUser(request: request)
         : throw _pluginNotAddedException("Auth");
   }
 
+  /// Returns [<List<AuthUserAttribute>>] or a list of a plugin-specific subclass.
+  ///
+  /// Throws an [AuthException], or possibly a subclass such as [SignedOutException].
+  /// Other AuthException subclasses are possible as a result of network issues,
+  /// misconfiguration, internal service errors or other less common reasons.
   Future<List<AuthUserAttribute>> fetchUserAttributes({AuthUserAttributeOptions options}) {
     var request = AuthUserAttributeRequest(options: options);
     return plugins.length == 1
         ? plugins[0].fetchUserAttributes(request: request)
         : throw _pluginNotAddedException("Auth");
   }
 
+  /// Returns [AuthSession] or a plugin-specific subclass.
```

Comment:
How do we best convey that customers **need** to cast it to a plugin specific class else it won't work? I think it's important since I've seen users struggling to figure out how to access the tokens. Same goes for other APIs as well where the cast is mandatory.

Reply:
It's tricky because I don't want to go into too much plugin-specific detail here since this documentation should broadly define the category behavior.